src/tests/decider-distro-tests-rs/tests/test.rs

```diff
@@ -119,3 +533,1006 @@ async fn test_decider_installed() {
+            assert_eq!(method, "eth_blockNumber");
+            send_response.send(json!("0x200")).unwrap();
+        }
+        // no new deals
+        {
+            let (method, _) = recv_request.recv().await.unwrap();
+            assert_eq!(method, "eth_getLogs");
+            send_response.send(json!([])).unwrap();
+        }
+    }
+
+    {
+        let (method, params) = recv_request.recv().await.unwrap();
+        assert_eq!(method, "eth_getLogs");
+        let log = serde_json::from_value::<LogsReq>(params[0].clone()).unwrap();
+        println!("{:?}", log);
+        assert_eq!(
+            log.address, deal.deal_id,
+            "wrong deal_id in the update-deal request"
+        );
+
+        let response = json!( [
+              {
+                "address": deal.deal_id,
+                "topics": [
+                  "0xc820a66d3bdd50a45cf12cda6dc8ec9e94fb5123edd7da736eea18316f8523a0"
+                ],
+                "data": "0x01551220000000000000000000000000000000000000000000000000000000004dd7ac07e70a485ee9f9a8e77736e526b1b20ffb911a508643c1c35bb2bdce16",
+                "blockNumber": "0x300",
+                "transactionHash": "0xb825edf7da59840ce838a9ed70aa0aa6c54c322ca5d6f0be4f070766e46ebbd8",
+                "transactionIndex": "0xb",
+                "blockHash": "0x34ba65babca6f1ef44da5f75c7bb4335c7b7484178a74003de5df139ac6551ed",
+                "logIndex": "0x26",
+                "removed": false
+              }
+            ]
+        );
+        send_response.send(response).unwrap();
+    }
+
+    wait_decider(&mut client).await;
+
+    server.shutdown().await
+}
+*/
+
+///
+/// Test how *Decider* calculates the block to poll.
+/// Block numbers sequence:
+/// 1. `0x0`  -- Decider should be able to poll from the beginning and don't break,so
+///            saved last_seen_block should be `0x0`
+///            In the eth_getLogs request: we can check that fromBlock is `0x0`
+///
+/// 2. `0x10` -- the number is less then the range decider polls
+///            Decider should move it's left boundary to this block, so
+///            saved last_seen_block should be `0x10`
+///            In the eth_getLogs request: we can check that fromBlock is `0x1`
+///
+/// 3. `0x10` -- again the same number in not very realistic case when Decider is too fast and the chain is too slow
+///            Decider shouldn't move it's left boundary anywhere, so
+///            saved last_seen_block should be `0x10`
+///            In the eth_getLogs request: we can check that fromBlock is `0x11`
+///
+/// 4. `0xffffff` -- big number which Decider shouldn't be able to process during one iteration
+///            Decider should move to some boundary configured boundary, which is 500 blocks, so
+///            saved last_seen_block should be `0x205`
+///            In the eth_getLogs request: we can check that fromBlock is `0x11`
+///            Note: the test depend on the knowledge that the range is 500 blocks,
```

Comment:
Sounds more and more that this number should be part of `init data`. WDYT?

Reply:
probably, but it feels like the idea for a separate refactoring pr

Reply:
can you create a task for that plz?

Reply:
https://linear.app/fluence/issue/NET-574/make-a-range-of-blocks-for-polling-in-decider-configurable